src/BookNavigator/bookmarks/ia-bookmarks.js

```diff
@@ -229,21 +229,23 @@ class IABookmarks extends LitElement {
     return bookmark;
   }
 
-  fetchBookmarks() {
-    return this.api.getAll().then(res => res.text()).then((res) => {
-      let response;
+  async fetchBookmarks() {
+    try {
+      const res = await this.api.getAll();
+      const resText = await res.text();
+      let parsedResponse;
       try {
-        response = JSON.parse(res);
+        parsedResponse = await JSON.parse(resText);
```

Comment:
`JSON.parse` doesn't return a promise, so no need for `await` here.

```suggestion
        parsedResponse = JSON.parse(resText);
```